File: src/main/java/com/kerosenelabs/billtracker/model/external/response/GoogleOAuthUserInfoResponse.java
package com.kerosenelabs.billtracker.model.external.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class GoogleOAuthUserInfoResponse {
    private String sub;
    private String name;

    @JsonProperty("given_name")
    private String givenName;

    @JsonProperty("family_name")
    private String familyName;

    private String email;
    private String picture;

    @JsonProperty("email_verified")
    private boolean emailVerified;
}

File: src/main/java/com/kerosenelabs/billtracker/model/external/response/GoogleOAuthUserInfoResponse.kt
package com.kerosenelabs.billtracker.model.external.response

class GoogleOAuthUserInfoResponse(
    val sub: String,
    val name: String,
    val givenName: String,
    val familyName: String,
    val email: String,
    val picture: String,
    val emailVeriied: Boolean
)
